quimb/tensor/optimize_pytorch.py

```diff
@@ -168,20 +278,42 @@ def optimize(self, max_steps, max_time=None):
         pbar = tqdm.tqdm(total=max_steps, disable=not self.progbar)
         self._maybe_start_timer(max_time)
         try:
-            for _ in range(max_steps):
+            for it in range(max_steps):
+                if self.progbar is False:
+                    t0 = time.time()
+
+                # keep the tn actually normalized during the optimization
+                # by doing STRICTLY IN-PLACE modification of tn_opt
+                if self.norm_fn_scalar is not None:
+                    torch, _ = _TORCH_DEVICE
+                    with torch.no_grad():
+                        fac = self.norm_fn_scalar(self.tn_opt)
+                        for tensor in self.tn_opt:
+                            tensor._data *= fac
+
                 self.optimizer.step(self.closure)
                 pbar.set_description(f"{self.loss}")
                 pbar.update()
                 self._n += 1
 
-                # check if there is a target loss we have reached
+                if self.progbar is False:
```

Comment:
I think this might be better as a separate non-default option like ``if self.progbar == 'simple'``. Having a silent option is pretty useful. 

Might also (at some later point) be worth storing the losses and times ``self.losses`` / ``self.iter_times`` etc for programmatic access rather than printing as well.